centrosome/filter.py

```diff
@@ -493,20 +496,59 @@ def roberts(image, mask=None):
     result[big_mask] = np.sqrt(diagonal*diagonal + anti_diagonal*anti_diagonal)
     return result
 
-def sobel(image, mask=None):
-    '''Calculate the absolute magnitude Sobel to find the edges
 
-    image - image to process
-    mask - mask of relevant points
+__u = np.array([[[1, 2, 1]]])
+__v = np.array([[[1, 0, -1]]])
 
-    Take the square root of the sum of the squares of the horizontal and
-    vertical Sobels to get a magnitude that's somewhat insensitive to
-    direction.
 
-    Note that scipy's Sobel returns a directional Sobel which isn't
-    useful for edge detection in its raw form.
-    '''
-    return np.sqrt(hsobel(image,mask)**2 + vsobel(image,mask)**2)
+__yt = (0, 2, 1)
+__zt = (2, 0, 1)
+
+
+def sobel(data, mask=None):
+    if data.ndim is 2:
+        return np.sqrt(hsobel(data, mask)**2 + vsobel(data, mask)**2)
+
+    result = np.sqrt(sobel_x(data, mask)**2 + sobel_y(data, mask)**2 + sobel_z(data, mask)**2)
+    result /= np.sqrt(3)
+    return result
+
+
+def sobel_x(data, mask=None):
+    data = skimage.img_as_float(data)
+    kernel = (__u.transpose(__zt) * __v * __u.transpose(__yt)).astype(float) / 16.0
```

Comment:
Normalization. In 2D it is 4.0, so we square it in 3D to get 16.0. Ensures result is in [0, 1].